src/CircuitSimulatorExecutor.cpp

```diff
@@ -0,0 +1,38 @@
+#include "CircuitSimulatorExecutor.hpp"
+
+CircuitSimulatorExecutor::CircuitSimulatorExecutor(
+    std::unique_ptr<SimulationTask> simulationTask) {
+  mTask   = std::move(simulationTask);
+  auto qc = std::make_unique<qc::QuantumComputation>(mTask->getQc()->clone());
+  mCircuitSimulator = std::make_unique<CircuitSimulator<>>(std::move(qc));
```

Comment:
After I move these into the execute function, where should the start of the clock be? Should it be after these object initializations (hence only recording the time of execution) or should it include everything?

Reply:
Good question.
Why not track both? That is, construction time and execution time?
For example, I would expect the construction time to get lower once the package data structures can be dynamically resized since a smaller allocation will be required in the beginning.